Py3stage1

degv364 opened the pull request:
PEP8 compliant, futurized, and reoganized repo.

degv364 committed:
Make futurized files PEP8 compliant

degv364 committed:
Reorganize project structure

degv364 committed:
Fix Pep8 issues after futurizing and reorganizing

degv364 committed:
Fix installation issues

jprltsnz commented on lib/mobile_robot_simulator/robot_models/__init__.py:
```diff
@@ -16,10 +16,13 @@
-import vectorfield
-def main():
-    pass
+from .differential_robot import Differential_robot
```
add `from __future__ import absolute_import` before this relative imports, please

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ackerman_robot.py:
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads  Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others  Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

degv364 replied:
The one that includes UCR is prefered. Thanks for the observation.

jprltsnz commented on README.md:
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+Copyright (c) 2016-2018 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
```
Sometimes this reads  Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others  Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ackerman_robot.py:
```diff
@@ -16,21 +16,24 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-#ackerman robot Class.
+# Ackerman robot Class.
```
This comment should be the Ackerman robot class's docstring: https://www.python.org/dev/peps/pep-0257/

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ackerman_robot.py:
```diff
@@ -16,21 +16,24 @@
     def __init__(self):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ackerman_robot.py:
```diff
@@ -16,21 +16,24 @@
     def set_params(self, params):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/__init__.py:
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ball_bot.py:
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ball_bot.py:
```diff
@@ -15,22 +15,24 @@
     def __init__(self):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/ball_bot.py:
```diff
@@ -15,22 +15,24 @@
     def set_params(self, params):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# Differential Robot Class.
```
This comment should be the Differential robot class's docstring: https://www.python.org/dev/peps/pep-0257/

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/omnidirectional_robot.py:
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/vectorfield/__init__.py:
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/vectorfield/points.py:
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory,
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on scripts/core.py:
```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on scripts/tools/goal_obstacle.py:
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# Copyright (c) 2016 Autonomous Robots and Cognitive Systems Laboratory.
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on setup.py:
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/omnidirectional_robot.py:
```diff
@@ -15,21 +15,24 @@
     def __init__(self):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/omnidirectional_robot.py:
```diff
@@ -15,21 +15,24 @@
     def set_params(self, params):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/walker.py:
```diff
@@ -15,21 +15,24 @@
     def __init__(self):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/walker.py:
```diff
@@ -15,21 +15,24 @@
     def set_params(self, params):
-        print "not implemented"
-        
+        print("not implemented")
```
change this to:
``` python
raise NotImplementedError()
```

jprltsnz commented on lib/mobile_robot_simulator/robot_models/walker.py:
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica
+# Copyright (c) 2011 Autonomous Robots and Cognitive Systems Laboratory
```
Sometimes this reads Autonomous Robots and Cognitive Systems Laboratory, Universidad de Costa Rica and others Autonomous Robots and Cognitive Systems Laboratory; without the Universidad de Costa Rica. Standarize to one or the other please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/omnidirectional_robot.py:
```diff
@@ -15,21 +15,24 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-#omnidirectional robot Class.
+# Omnidirectional robot Class.
```
Again, turn this into a docstring, please.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/walker.py:
```diff
@@ -15,21 +15,24 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-#Walker robot Class.
+# Walker robot Class.
```
Docstring this is.

jprltsnz commented on setup.py:
```diff
@@ -14,16 +14,71 @@
+
+setup(
+    name='mobile_robot_simulator',
+    version='0.1.1',
```
Change to: ` version=find_version('lib/arcospyu/__init__.py')` please.

degv364 replied:
@jcrono  I will change it to 'lib/mobile_robot_simulator/__init__.py' or do you want a single version for the entire simulator? (and use that single arcospyu file as the reference?)

jprltsnz commented on requirements.txt:
```diff
@@ -0,0 +1,22 @@
+########################
```
None of this are actual requirements. All of this are for testing/QA please rename this file as `requirements.dev.txt`

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+        self.robot_center = array([0., 0., 0])
+        self.robot_planes = [
+            # normal point
+            [(1., 0., 0), (old_div(self.robot_dim[0], 2.), 0., 0.)],
```
old_div is not required here since this is a floating point division it will work consistently though python2 and python3. Please change to regular `/` division operator

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+        self.robot_center = robot_center
+        self.robot_planes = [
+            # normal point
+            [(1., 0., 0), (old_div(self.robot_dim[0], 2.), 0., 0.)],
```
old_division is not requiered here either.

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+    def wheel_velocity(self, descentralized_velocity, angle):
+        # Compute wheel velocities from the required velocity of the
+        # descentralized point, and the current angle of the robot.
+        right_wheel = old_div(((self.d * cos(angle) - 0.5 * self.b *
```
Same here, no old_div

jprltsnz commented on lib/mobile_robot_simulator/robot_models/differential_robot.py:
```diff
@@ -0,0 +1,115 @@
+                               (self.d * sin(angle) + 0.5 * self.b *
+                                cos(angle)) * descentralized_velocity[1]),
+                              (self.d * self.r))
+        left_wheel = old_div(((self.d * cos(angle) + 0.5 * self.b
```
And here, no old_div

jprltsnz commented on Makefile:
```diff
@@ -17,7 +17,8 @@
 
```
This makefile has un runable rules like config, purge and %.deb either fix them if requiered or remove them

jprltsnz reviewed (changes requested):
Please add a coding comment at the top of each python file: `# -*- coding: utf-8 -*-`

Also remove the `#! /usr/bin/python` from all non-script files, they are not meant to be executed.

Great job, thanks.

degv364 committed:
Fix file headers

Remove '#! /usr/bin/python' from non-scripts files
Add # -*- coding: utf-8 -*- at the top of every file
Add UCR to the license header

degv364 committed:
Add docstrings PEP 257 compliant

degv364 committed:
Raise not implemented error in robots other than differential

degv364 committed:
Handle version at lib/mobile_robot_simulator/__init__.py

degv364 committed:
Rename requirements to requirements.dev

degv364 committed:
Remove the old_div calls, as they are not required

degv364 committed:
Fis installation issues due to unfound modules

degv364 commented:
@jcrono Thanks for your comments. They have been addressed.

jprltsnz commented on scripts/tools/goal_obstacle.py:
```diff
@@ -0,0 +1,175 @@
+                self.obstacle_vel_list[i][:2] += 0.6 * random(2) - array(
+                    [0.3, 0.3])
+                if norm(self.obstacle_pose_list[i][:2, 3]) > 12:
+                    self.obstacle_vel_list[i][:2] += old_div(
```
old_div not required, please use the regular division operator `/`

degv364 replied:
Good catch. Now all files are PEP238 compliant.

jprltsnz reviewed (changes requested):
I missed a `old_div` fix that, and this looks good to me. Great Job

degv364 committed:
Remove broken rules from Makefile

degv364 committed:
Make all files PEP238 compliant
